internal/mining/mining.go

```diff
@@ -1526,48 +1596,53 @@ mempoolLoop:
 			}
 		}
 
+		if miningView.IsRejected(prioItem.tx.Hash()) {
+			// If the transaction or any of its ancestors have been rejected,
+			// discard the transaction.
+			continue
+		}
+
+		ancestors := miningView.Ancestors(tx.Hash())
+		ancestorStats, _ := miningView.AncestorStats(tx.Hash())
+
+		prioItem.feePerKB = calcFeePerKb(prioItem.txDesc, ancestorStats)
+
 		// Enforce maximum block size.  Also check for overflow.
 		txSize := uint32(tx.MsgTx().SerializeSize())
-		blockPlusTxSize := blockSize + txSize
+		blockPlusTxSize := blockSize + txSize + uint32(ancestorStats.SizeBytes)
 		if blockPlusTxSize < blockSize ||
 			blockPlusTxSize >= g.policy.BlockMaxSize {
 			log.Tracef("Skipping tx %s (size %v) because it "+
 				"would exceed the max block size; cur block "+
 				"size %v, cur num tx %v", tx.Hash(), txSize,
 				blockSize, len(blockTxns))
 			logSkippedDeps(tx, deps)
+			miningView.Reject(tx.Hash())
 			continue
 		}
 
 		// Enforce maximum signature operations per block.  Also check
 		// for overflow.
-		numSigOps := int64(blockchain.CountSigOps(tx, false, isSSGen,
-			isTreasuryEnabled))
-		if blockSigOps+numSigOps < blockSigOps ||
-			blockSigOps+numSigOps > blockchain.MaxSigOpsPerBlock {
+		numSigOps := int64(prioItem.txDesc.NumSigOps) + ancestorStats.NumSigOps
+		numSigOpsBundle := numSigOps + ancestorStats.NumSigOps
```

Comment:
Why is `ancestorStats.NumSigOps`being added here if its already been added to `numSigOps` above?

Reply:
Updated